go/launchd/launchd.go

```diff
@@ -42,43 +48,71 @@ func (s Service) Label() string { return s.label }
 
 // Plist defines a launchd plist
 type Plist struct {
-	label   string
-	binPath string
-	args    []string
-	envVars map[string]string
+	label     string
+	binPath   string
+	args      []string
+	envVars   map[string]string
+	keepAlive bool
 }
 
 // NewPlist constructs a launchd service.
 func NewPlist(label string, binPath string, args []string, envVars map[string]string) Plist {
 	return Plist{
-		label:   label,
-		binPath: binPath,
-		args:    args,
-		envVars: envVars,
+		label:     label,
+		binPath:   binPath,
+		args:      args,
+		envVars:   envVars,
+		keepAlive: true,
 	}
 }
 
-// Load will load the service.
-// If restart=true, then we'll unload it first.
-func (s Service) Load(restart bool) error {
-	// Unload first if we're forcing
+// Load will start the service.
+func (s Service) Start() error {
 	plistDest := s.plistDestination()
-	if restart {
-		exec.Command("/bin/launchctl", "unload", plistDest).Output()
-	}
-	fmt.Fprintf(s.writer, "Loading %s\n", s.label)
+	s.info("Starting %s", s.label)
+	// We start using load -w on plist file
 	_, err := exec.Command("/bin/launchctl", "load", "-w", plistDest).Output()
 	return err
 }
 
-// Unload will unload the service
-func (s Service) Unload() error {
-	plistDest := s.plistDestination()
-	fmt.Fprintf(s.writer, "Unloading %s\n", s.label)
-	_, err := exec.Command("/bin/launchctl", "unload", plistDest).Output()
+// Stop a service.
+func (s Service) Stop(wait bool) error {
+	s.info("Stopping %s", s.label)
+	// We stop by removing the job. This works for non-demand and demand jobs.
+	_, err := exec.Command("/bin/launchctl", "remove", s.label).Output()
+	if wait {
+		err = s.WaitForExit()
+		if err != nil {
+			return err
+		}
+	}
 	return err
 }
 
+// WaitForExit waits for service to exit
+func (s Service) WaitForExit() error {
+	running := true
+	// Timeout after 5 seconds
+	for i := 0; i < 5; i++ {
+		status, err := s.LoadStatus()
+		if err != nil {
+			return err
+		}
+		if status == nil || !status.IsRunning() {
+			running = false
+			break
+		}
+		if (i+1)%3 == 0 {
```

Comment:
Yeah was trying to not log when i == 0.... changed this in the next commit though so it's moot